catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/Context.java

```diff
@@ -54,4 +65,14 @@ public void setParentMetacard(Metacard parentMetacard) {
         this.parentMetacard = Optional.of(parentMetacard);
     }
 
+    /**
+     * When updating the parent metacard or uploading more child metacards, lock this return
+     * value.
+     *
+     * @return true if parent needs to be updated
+     */
+    public AtomicBoolean isParentDirty() {
```

Comment:
interesting ... something like a method called 'void modifyParent(Consumer)' that synchronizes internally and sets the dirty flag. And probably a second method that indicates the parent is no longer dirty, or a boolean flag that indicates such on 'modifyParent'